crypto/dh/dh_pmeth.c

```diff
@@ -114,11 +114,21 @@ static int pkey_dh_init(EVP_PKEY_CTX *ctx)
     return 1;
 }
 
+static void pkey_dh_cleanup(EVP_PKEY_CTX *ctx)
+{
+    DH_PKEY_CTX *dctx = ctx->data;
+    if (dctx) {
```

Comment:
yes, so in case the cleanup function gets called twice it will be "safe"  like free(0)